bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -153,6 +154,21 @@ pub struct Client {
         Arc<tokio::sync::RwLock<Option<SessionVerificationController>>>,
 }
 
+impl Drop for Client {
+    fn drop(&mut self) {
+        // HACK: the Client holds references to deadpool's pools, which require to be in
+        // a tokio Runtime because their `Drop` impl will use `block_on`. Now, we can't
+        // just drop the stores without breaking all the abstractions, so what
+        // we do instead is that we replace the entire client with a new dummy
+        // one, configured with in-memory stores; that will drop the previous
+        // one.
+        RUNTIME.block_on(async move {
+            let url = self.inner.homeserver();
+            self.inner = MatrixClient::builder().homeserver_url(url).build().await.unwrap();
```

Comment:
Hmm, I'm a bit worried that we're going to accidentally add some requests to the `build()` here, currently we don't trigger any requests because we set the `homeserver_url()`.

Though not sure how to ensure that this does not happen.